src/io/directory.js

```diff
@@ -0,0 +1,83 @@
+import * as path from 'path';
+import { createReadStream } from 'fs';
+
+import { IOBase } from 'io/base';
+import { walkPromise } from 'io/utils';
+
+import { singleLineString } from '../utils';
+
+import log from 'logger';
+
+
+export class Directory extends IOBase {
+
+  constructor(dirPath) {
+    super();
+    this.dirPath = dirPath;
+    this.files = {};
+    this.entries = [];
+  }
+
+  getFiles(_walkPromise=walkPromise) {
+    // If we have already processed this directory and have data
+    // on this instance return that.
+    if (Object.keys(this.files).length) {
+      log.info(singleLineString`Files already exist for directory
+               "${this.dirPath}" returning cached data`);
+      return Promise.resolve(this.files);
+    }
+    return _walkPromise(this.dirPath)
+      .then((files) => {
+        this.files = files;
+        this.entries = Object.keys(files);
+        return files;
+      });
+  }
+
+  getFileAsStream(relativeFilePath) {
+
+    if (!this.files.hasOwnProperty(relativeFilePath)) {
+      return Promise.reject(
+        new Error(`Path "${relativeFilePath}" does not exist in this dir.`));
+    }
+
+    var absDirPath = path.resolve(this.dirPath);
```

Comment:
Nitpick, but we use `relativeFilePath` above, can we use `absoluteDirPath` here for clarity?